### PR TITLE
[Whisper]: Fix attention mask length for NPU

### PR DESCRIPTION
**Pull request by as-suvorov**
## Description

CVS-###

Fixes #(issue)

## Checklist:
- [x] Tests have been updated or added to cover the new code. 
- [ ] This patch fully addresses the ticket. 
- [ ] I have made corresponding changes to the documentation.

### CODE REVIEW

File: tests/python_tests/test_whisper_pipeline_static.py [hunk 1]
```diff
@@ -91,6 +101,15 @@ def compare_results_with_assert(expected, actual_out):
         assert expected.texts[i] == actual_out.texts[i]
 
 
+def compare_word_timestamps_results_with_assert(expected, actual_out, ts_tolerance=0.06):
+    assert len(expected.words) == len(actual_out.words)
+
+    for exp_word, act_word in zip(expected.words, actual_out.words):
+        assert exp_word.word == act_word.word
+        assert round(abs(exp_word.start_ts - act_word.start_ts), 2) <= ts_tolerance
+        assert round(abs(exp_word.end_ts - act_word.end_ts), 2) <= ts_tolerance
```
Comment (Wovchena) [inline tests/python_tests/test_whisper_pipeline_static.py]: I learnt that there's `pytest.approx`. Use it

File: src/cpp/src/whisper/pipeline_static.cpp [hunk 2]
```diff
@@ -1063,17 +1063,17 @@ WhisperPipeline::StaticWhisperPipeline::StaticWhisperPipeline(const std::filesys
         add_cache_position_input(decoder_with_past_model);
     }
 
-    add_attention_mask_input(decoder_model, true /* transform_cross_attn */, last_hidden_state_shape[1].get_length());
+    const size_t max_sequence_length = 448;
+
+    // When word_timestamps is enabled decoder can receive tokens decoded from entire audio chunk which is 447
```
[the remark this answers is not in the document]
Comment (Wovchena) [reply src/cpp/src/whisper/pipeline_static.cpp]: At least `,` is a reasonable note :)
